README.md

```diff
@@ -0,0 +1,353 @@
+
+If the alternative chain has the same block ID in the `head_block` and `last_common_block` fields, and the `length` is zero, that means the node on the fork is stalled and doesn’t generate or accept any blocks.
+
+For every node that has ever connected, there is a link to the last block received from that node, called `leash`. When receiving updates from a node, Fork Detector updates the `leash` for that node. For disconnected nodes, the `leash` points to the last received block.
+
+### Block graph initialization
+
+On the first run, Fork Detector builds the graph of blocks starting from the Genesis block in the same way as a node does. During the initial graph build, the number of alternative chains is minimal.
+
+### Version selection
+
+One must provide a list of protocol versions supported by Fork Detector. Connecting to nodes with different versions allows the detection of alternative chains that may arise due to lag or the inability of some generators to update during a Waves network upgrade.
+When establishing an outgoing connection to a new node, Fork Detector attempts to handshake with each version from the list until a successful connection is made. For incoming connections from unknown nodes, Fork Detector responds with the version received from the node if that version is included in the list.
+
+## HTTP API Methods
+
+### `GET` `/api/peers/all`
+
+The method returns a list of addresses of all known nodes. These addresses were received from other nodes during the known node exchange, but a connection to all of them is not guaranteed.
+
+Reply example:
+```json
+[
+  {
+    "address": "79.199.35.153:0",
+    "nonce": 0,
+    "name": "",
+    "version": "0.0.0",
+    "state": 0,
+    "next_attempt": "0001-01-01T00:00:00Z",
+    "score": null
+  }
+]
+```
+
+Reply fields:
+* `address` - IP address and port of the node. The port has a zero value in case of an incoming connection (node on an ephemeral port) or if no connection was established.
+* `nonce` - random nonce; zero if there was no connection to the node.
+* `name` - name of the node; empty string if no connection was made.
+* `version` - node’s version; zero if no connection was established.
+* `state` - state of the connection to the node:
+    * `0` `Unknown`- unknown node, no successful connection was made to the address.
+    * `1` `Connected` - successful connection was made, and the network byte and version are acceptable.
+    * `2` `Hostile` - successful connection was established, but the network byte or version is not acceptable.
+* `next_attempt` - time when the next attempt to connect to the node is planned. This field is used during version selection (see the section [Version Selection](#version-selection)).
+* `score` - last known value of the Score; zero if there was no connection.
```

Comment:
I see `null` value in example. So `null` or `0` value if there was no connection?

Reply:
Yep, `null` is correct, because it's a BigInt value.